flows/runs/run.go

```diff
@@ -163,10 +163,29 @@ func (r *flowRun) SetWebhook(rr *utils.RequestResponse) {
 func (r *flowRun) Extra() utils.JSONFragment         { return r.extra }
 func (r *flowRun) SetExtra(extra utils.JSONFragment) { r.extra = extra }
 
+func (r *flowRun) ExpiresOn() *time.Time { return r.expiresOn }
+func (r *flowRun) ResetExpiration(from *time.Time) {
+	if r.Flow().ExpireAfterMinutes() != nil {
+		if from == nil {
+			now := time.Now().UTC()
+			from = &now
+		}
+
+		expiresAfterMinutes := time.Duration(*r.Flow().ExpireAfterMinutes())
+		expiresOn := from.Add(expiresAfterMinutes * time.Minute)
+
+		r.expiresOn = &expiresOn
+		r.setModifiedOn(time.Now().UTC())
```

Comment:
`from` is sometimes a child run's expiredOn so wouldn't make sense to use for modifiedOn.

There is an argument for the engine not maintaining createdOn or modifiedOn and letting those be the db times on the caller side. That's probably what we need if we want run syncing over the API to still work properly.